src/core/blob.cpp

```diff
@@ -20,31 +26,37 @@ blob::blob(session & s)
 
 blob::~blob()
 {
-    delete backEnd_;
+    if (backEnd_)
+        delete backEnd_;
 }
 
 std::size_t blob::get_len()
 {
+    assert(backEnd_);
```

Comment:
I'd strongly prefer to avoid reintroducing asserts in SOCI code. In the case of this class, we should throw an exception if `backEnd_` is null from the ctor ensuring that this is never the case in any method of this class.